src/ossl/ecc.rs

```diff
@@ -709,3 +715,156 @@ impl Verify for EccOperation {
         Ok(self.output_len)
     }
 }
+
+fn new_pkey_ctx_from_obj(pkey: &mut EvpPkey) -> KResult<EvpPkeyCtx> {
```

Comment:
I think you need to use lifetimes here, otherwise the pkey may be deallocated while you are still holding a reference to it.

Rust can't see that you are because this is done through unsafe constructs in the C function being called.

Arguably lifetimes should be also used in EvpPkeyCtx::from_ptr(), I will look into that later (opening an issue).

Reply:
Good point. I did not work with the lifetimes much yet so I am not sure how to easily approach it now, given that the contexts are all defined in